[ADD] 'pos_product_template'

legalsylvain opened the pull request:
Propose a module to select variant of template.

Main features:
- products variant are now hidden from the POS (only one template is displayed);
- it is still possible to scan variant;
- an extra screen is displayed to select variant when a template is selected;

see https://www.youtube.com/watch?v=sCZr39NxSlU&feature=youtu.be

Review and test are welcome.

legalsylvain committed:
[ADD] 'pos_product_template'

legalsylvain committed:
[IMP] better display of attributes values;
[FIX] overflow bug;

legalsylvain committed:
[ADD] fr translation;

sebastienbeau commented:
:+1: Great job Sylvain, I test it and it's work perfectly ! Thanks

sylvainc commented on pos_product_template/static/src/css/ppt.css:
```diff
@@ -0,0 +1,206 @@
+
+
+/*  ********* Variant Display  ********************************************* */
+/* This part is a copy past from Odoo Project, with some values changed      */
```
typo: s/past/paste
What about using mulitple classes (class="product variant") in the QWeb template and ".pos .product.variant" in the CSS, and just redefine the attributes you need ?

sylvainc commented on pos_product_template/static/src/js/ppt.js:
```diff
@@ -0,0 +1,565 @@
+
+        init: function(parent, options) {
+            this._super(parent,options);
+            this.template_name = '';
```
this.template_name seems unsued

sylvainc commented on pos_product_template/static/src/js/ppt.js:
```diff
@@ -0,0 +1,565 @@
+            var template = this.pos.db.template_by_id[product_tmpl_id];
+
+            // Display Name of Template
+            this.template_name = template.name;
```
same here: this.template_name seems unsued

sylvainc commented on pos_product_template/static/src/js/ppt.js:
```diff
@@ -0,0 +1,565 @@
+            this.product_template = null;
+            this.click_set_attribute_handler = function(event){
+                /*TODO: Refactor this function with elegant DOM manipulation */
+                // remove selected item
```
Maybe using JQuery could help

legalsylvain replied:
Yes, I tried, but I didn't manage with JQuery. But I'm not JQuery Expert. Maybe could we accept this PR like that, as it's not a bug but only a 'not very clear code'. What do you think about that point ?

sylvainc replied:
Hi, it's ok for me.
Thanks!

sylvainc commented:
Well done!

AdilHoumadi commented:
Very nice, good job :+1:

pedrobaeza commented:
Hi, Sylvain, great module!

Can you please provide pot translation file? Addressing this and @sylvainc's issues, I will merge it.

legalsylvain committed:
[REF] remove useless variable, courtesy to Sylvain C;

legalsylvain committed:
[FIX] Typo

legalsylvain committed:
[ADD] 'pos_product_template' pot file

legalsylvain committed:
[ADD] Add extra demo data that generate display bug

legalsylvain committed:
[FIX] hide product attribute value in POS if value has no product associated to it

legalsylvain committed:
[IMP] description of the module in openerp.py file;

legalsylvain committed:
[REF]

legalsylvain committed:
[IMP] Add screenshot descrption;

legalsylvain commented:
Hi all, 
i just 
- Fixed a bug that @sebastienbeau said me: There was a bug that displayed "value : 0 products" if there was no product with this value for the current template;See extra-demo data added: All the IPad has the same value "2.4Ghz" for the attribute "Wifi"; So display other values are not relevant;
- added some extra-description (comment and screenshot of the module);

Regards.

pedrobaeza commented:
Great, so I proceed with the merge and if any other improvement is found, we make another PR.
